src/clr-angular/data/datagrid/providers/state.provider.ts

```diff
@@ -59,13 +59,16 @@ export class StateProvider<T> {
     if (activeFilters.length > 0) {
       state.filters = [];
       for (const filter of activeFilters) {
-        if (filter instanceof DatagridStringFilterImpl) {
+        if (filter.state) {
+          state.filters.push(filter.state);
+          continue;
+        } else if (filter instanceof DatagridStringFilterImpl) {
```

Comment:
This if should be removed, instead just implement `state` and `equals()` in our built-in filters. It'll simplify everything.

Reply:
Done